modules/ipc/include/hephaestus/ipc/zenoh/session.h

```diff
@@ -18,9 +19,40 @@ namespace heph::ipc::zenoh {
 enum class Mode : uint8_t { PEER = 0, CLIENT, ROUTER };
 enum class Protocol : uint8_t { ANY = 0, UDP, TCP };
 
+/// There are a lot of options to configure a zenoh session,
+/// See  for more information https://zenoh.io/docs/manual/configuration/#configuration-files
+struct ZenohConfig {
+  ZenohConfig() {
+    zconfig.insert_json5(Z_CONFIG_ADD_TIMESTAMP_KEY, "true");
+  }
+  explicit ZenohConfig(std::string const& path) : zconfig(::zenoh::Config::from_file(path)) {
+  }
+
+  ~ZenohConfig() noexcept = default;
+  ZenohConfig(const ZenohConfig&) = delete;
+  auto operator=(const ZenohConfig&) -> ZenohConfig& = delete;
+  ZenohConfig(ZenohConfig&&) noexcept = default;
+  auto operator=(ZenohConfig&&) noexcept -> ZenohConfig& = default;
+
+  ::zenoh::Config zconfig{ ::zenoh::Config::create_default() };
+  std::size_t cache_size{ 0 };
```

Comment:
ok so this is something where in the past I bent the rules for convenience:

this is not a session config, but of the publisher and subscriber, so given that we are polishing the interface, I would remove it from here